lightly/api/api_workflow_upload_dataset.py

```diff
@@ -40,11 +40,14 @@ def upload_dataset(self: ApiWorkflowClient, input: Union[str, LightlyDataset], m
                 full images, or metadata only.
 
         Raises:
-            ValueError if dataset is too large.
+            ValueError if dataset is too large or input has the wrong type
             RuntimeError if the connection to the server failed.
-            RuntimeError if dataset already has an initial tag.
 
         """
+        no_tags_on_server = len(self._get_all_tags())
+        if no_tags_on_server > 0:
+            warnings.warn(f"Dataset has already been uploaded to the platform. Skipping upload.")
```

Comment:
I'd maybe reformulate the warning. What about f"Dataset with id {dataset_id} has already been completed. Skipping upload."

Reply:
Also, not sure if we want to throw a warning or just print the message and return.

Reply:
I changed the warning text, but would keep the warning, as messages are easily overlook. And it is indeed a warning, as the user did not follow the workflow.